addons/product/static/src/js/product_configurator/product_template_attribute_line/product_template_attribute_line.js

```diff
@@ -0,0 +1,135 @@
+                    html_color: [Boolean, String], // backend sends 'false' when there is no color
+                    is_custom: Boolean,
+                    price_extra: { type: Number, optional: true},
+                    excluded: { type: Boolean, optional: true },
+                },
+            },
+        },
+        selected_attribute_value_ids: { type: Array, element: Number },
+        create_variant: {
+            type: String,
+            validate: type => ["always", "dynamic", "no_variant"].includes(type),
+        },
+        customValue: { type: String, optional: true },
+    };
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Update the selected PTAV in the state.
+     *
+     * @param {Event} event
+     */
+    updateSelectedPTAV(event) {
+        this.env.updateProductTemplateSelectedPTAV(
+            this.props.productTmplId, this.props.id, event.target.value, this.props.attribute.display_type == 'multi'
+        );
+    }
+
+    /**
+     * Update in the state the custom value of the selected PTAV.
+     *
+     * @param {Event} event
+     */
+    updateCustomValue(event) {
+        this.env.updatePTAVCustomValue(
+            this.props.productTmplId, this.props.selected_attribute_value_ids[0], event.target.value
+        );
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return template name to use by checking the display type in the props.
+     *
+     * Each attribute line can have one of this five display types:
+     *      - 'Color'  : Display each attribute as a circle filled with said color.
+     *      - 'Pills'  : Display each attribute as a rectangle-shaped element.
+     *      - 'Radio'  : Display each attribute as a radio element.
+     *      - 'Select' : Display each attribute in a selection tag.
+     *      - 'Multi'  : Display each attribute in a multi-checkbox tag.
+     *
+     * @return {String} - The template name to use.
+     */
+    getPTAVTemplate() {
+        switch(this.props.attribute.display_type) {
+            case 'color':
+                return 'product.ptav-color';
+            case 'multi':
+                return 'product.ptav-multi';
+            case 'pills':
+                return 'product.ptav-pills';
+            case 'radio':
+                return 'product.ptav-radio';
+            case 'select':
+                return 'product.ptav-select';
+        }
+    }
+
+    /**
+     * Return the name of the PTAV
+     *
+     * In the selection HTML tag, it is impossible to show the component `BadgeExtraPrice`. Append
+     * the extra price to the name to ensure that the extra price will be shown.
+     * Note: used in `productConfigurator.ptav-select`.
```

Comment:
Is is still the right name ?
```suggestion
     * Note: used in `productConfigurator.ptav-select`.
```